libs/user_extensions/src/TTAlpsEvent.cpp

```diff
@@ -125,3 +124,90 @@ bool TTAlpsEvent::ParticleHasISRmotherAfterTopMother(int particleIndex) {
 
   return ParticleHasISRmotherAfterTopMother(motherIndex);
 }
+
+bool TTAlpsEvent::IsGoodMuonFromALP(int muonIndex) {
+
+  auto genParticles = event->GetCollection("GenPart");
+  auto muon = asNanoGenParticle(genParticles->at(muonIndex));
+  
+  if (!muon->IsLastCopy()) return false;
+  if(!muon->IsMuon()) return false;
+
+  int motherIndex = muon->GetMotherIndex();
+  if (motherIndex < 0) return false;
+  auto mother = asNanoGenParticle(genParticles->at(motherIndex));
+
+  // loop over pythia copies of the muon
+  if (mother->IsMuon() == 13) {
+    while(mother->IsMuon() == 13) {
+      motherIndex = muon->GetMotherIndex();
+      if (motherIndex < 0) return false;
+      mother = asNanoGenParticle(genParticles->at(motherIndex));
+    }
+  }
+  // mother must be an ALP
+  if (!mother->IsLastCopy()) return false;
+  if (abs(mother->GetPdgId()) != 54) return false;
```

Comment:
Is 54 a code for an ALP? Please define a variable just to give this number a name (avoid "magic numbers")